ports/esp32s2/common-hal/wifi/Radio.c

```diff
@@ -142,6 +142,11 @@ wifi_radio_error_t common_hal_wifi_radio_connect(wifi_radio_obj_t *self, uint8_t
     } else {
         config->sta.bssid_set = 0;
     }
+    // If channel is 0 (default/unset) and BSSID is not given, do a full scan instead of fast scan
+    // This will ensure that the best AP in range is chosen automatically
+    if ((config->sta.bssid_set == 0) && (config->sta.channel == 0)) {
+        config->sta.scan_method = WIFI_ALL_CHANNEL_SCAN;
+    }
```

Comment:
Please set the FAST method in an else. Otherwise you won't be able to switch between the two.

Reply:
I'll update the PR this evening, but WIFI_FAST_SCAN would be the default anyway (behaviour as before the change).

```suggestion
    } else {
        config->sta.scan_method = WIFI_FAST_SCAN;
    }
```